src/cargo/util/config/mod.rs

```diff
@@ -1004,18 +1006,19 @@ impl Config {
     }
 
     pub(crate) fn load_values_unmerged(&self) -> CargoResult<Vec<ConfigValue>> {
+        self._load_values_unmerged()
```

Comment:
> The config code needing some changing around so that the checks could be placed in one location.

How bad would it be to split this up into multiple commits so we can more easily see this as smaller, cohesive sets of changes?  As of right now, I have to parse through everything to figure out what led to each change.

Reply:
Sure, I can look at splitting it up!

Reply:
OK, I reworked it into multiple commits.  Hopefully that can be a little easier to follow, as it is roughly the order that I implemented it.

Reply:
Thanks!  This has been a big help!